### PR TITLE
Bug 1186542, 1166448, 1185916, 1183536, 1177577; spot & golden userdata

### PR DESCRIPTION
**Pull request by grenade**
Migrated win 2008 userdata script to powershell to fix the bugs listed above.
This userdata script (2008.user_data) behaves more like cloud init in that it accepts tokens from cloud tools which are used to configure the instance. This is needed to support spot hostnaming, nxlog configuration and many other settings.
The new script:
 - logs output to the puppet mailing list, to enable debugging
 - checks and corrects the hostname
 - checks and corrects the primary dns suffix
 - checks and corrects nxlog aggregator config
 - ensures only a single puppet run on golden
 - handles reboots and shutdown

### CODE REVIEW

File: configs/2008.user_data [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  Write-Log -message 'deleting RunPuppet scheduled task' -severity 'INFO'
+  Unregister-ScheduledTask -TaskName 'RunPuppet' -Confirm:$false
+
+  Write-Log -message 'settting environment variables' -severity 'INFO'
+  [Environment]::SetEnvironmentVariable("FACTER_domain", "{domain}", "Process")
+  [Environment]::SetEnvironmentVariable("FACTER_hostname", "{hostname}", "Process")
+  [Environment]::SetEnvironmentVariable("FACTER_fqdn", ('{fqdn}'), "Process")
+  [Environment]::SetEnvironmentVariable("COMPUTERNAME", "{hostname}", "Machine")
+
+  Write-Log -message 'running puppettization script' -severity 'INFO'
```
Comment (djmitche) [inline configs/2008.user_data]: speling

File: configs/2008.user_data [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    [string] $subject
+  )
+  Send-MailMessage -To $global:logRecipients -Subject $subject -Body ([IO.File]::ReadAllText($logfile)) -SmtpServer $smtpServer -From ('{{0}}@{fqdn}' -f $env:UserName)
+}}
+
+# determines if the hostname is correctly set
+function Is-HostnameSetCorrectly {{
+  # todo: reinstate when we find a reliable way to set hostname
+  #Set-Hostname
+  #return $true
```
Comment (djmitche) [inline configs/2008.user_data]: I wouldn't be expecting an Is-* function to have side effects..

**Comment by djmitche** [comment]
:+1: I like it!

File: configs/2008.user_data (hunk 1 above)
Comment (grenade) [inline configs/2008.user_data]: hope i found them all...

File: configs/2008.user_data (hunk 2 above)
Comment (grenade) [inline configs/2008.user_data]: my bad. forgot to remove some lines i was debugging with.

**Pull request merged**